src/limiter.rs

```diff
@@ -312,6 +319,22 @@ impl<C: Clock> Limiter<C> {
         bucket.consume(size)
     }
 
+    /// Reverts the consumption of the given bytes size.
+    pub fn revert_consume(&self, byte_size: usize) {
+        self.total_bytes_consumed
+            .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |x| if x > byte_size {
+                Some(x - byte_size) } else { Some(0) }
```

Comment:
```rust
|x| Some(x.saturating_sub(byte_size))
```

Reply:
Looks like a heavy CAS loop underneath. Could we use `fetch_sub` instead? Assuming no reset is wrongly called between `consume` and `revert_consume`.

Reply:
I believe `revert_consume` is rarely called, so the `compare_exchange` loop underneath is acceptable. The assumption is not correct, cause the reverting may happen after the `total_bytes_consumed` is reset.